### PR TITLE
created Book entity, its service and repository

### PR DESCRIPTION
**Pull request by BlueVioletTeti**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/BookstoreApplication.java [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class BookstoreApplication {
+    @Autowired
+    private BookService bookService;
```
Comment (Denis-Balako) [inline src/main/java/mate/academy/bookstore/BookstoreApplication.java]: Autowire through constructor

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+@Table(name = "books")
```
Comment (Denis-Balako) [inline src/main/java/mate/academy/bookstore/model/Book.java]: https://medium.com/@centizennationwide/mysql-naming-conventions-e3a6f6219efe

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
```
Comment (Denis-Balako) [inline src/main/java/mate/academy/bookstore/model/Book.java:29]: Also, will don't respond to naming conventions

File: src/main/java/mate/academy/bookstore/repository/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        try {
+            entityManager = entityManagerFactory.createEntityManager();
+            transaction = entityManager.getTransaction();
+            transaction.begin();
+            entityManager.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
```
Comment (Denis-Balako) [inline src/main/java/mate/academy/bookstore/repository/BookRepositoryImpl.java:35]: Add an exception here

**Review by Denis-Balako — CHANGES_REQUESTED** [review]
Good job! Let's improve it!

File: pom.xml [hunk 5]
```diff
@@ -28,6 +28,35 @@
...
+		<dependency>
+			<groupId>com.mysql</groupId>
+			<artifactId>mysql-connector-j</artifactId>
+			<version>8.1.0</version>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+			<version>3.0.6</version>
```
Comment (vovasalyha) [inline pom.xml:51]: versions for all starters are supplied by parent pom

File: src/main/java/mate/academy/bookstore/model/Book.java (hunk 3 above)
Comment (vovasalyha) [reply src/main/java/mate/academy/bookstore/model/Book.java:29]: Actually, Hibernate by default resolves it in snake_case, but let it be.

**Review by vovasalyha — APPROVED** [review]
GJ. Take a look on few suggestions.

**Pull request merged**